core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1049,6 +1052,7 @@ object KafkaConfig {
       .defineInternal(BrokerHeartbeatIntervalMsProp, INT, Defaults.BrokerHeartbeatIntervalMs, null, MEDIUM, BrokerHeartbeatIntervalMsDoc)
       .defineInternal(BrokerSessionTimeoutMsProp, INT, Defaults.BrokerSessionTimeoutMs, null, MEDIUM, BrokerSessionTimeoutMsDoc)
       .defineInternal(MetadataLogDirProp, STRING, null, null, HIGH, MetadataLogDirDoc)
+      .defineInternal(MetadataSnapshotMinNewRecordBytesProp, INT, Defaults.MetadataSnapshotMinNewRecordBytes, atLeast(1), HIGH, MetadataSnapshotMinNewRecordBytesDoc)
```

Comment:
This should certainly be a LONG, not INT. I could certainly imagine a snapshot bigger than 2GB (although hopefully that won't be the norm for a while)

Reply:
Sounds good to me. I wasn't sure that we wanted to support for this value to be greater than `Int.MaxValue`.